### ISSUE TITLE
Depend on external W3C propagation library

### ISSUE THREAD

**Issue by jmacd** [open-telemetry/opentelemetry-go#391]
The proposal is to replace the implementation of the W3C `traceparent` context to use code from https://github.com/lightstep/tracecontext.go, which would be copied into a new OpenTelemetry- or W3C-organized GitHub repository. The primary motivation is -- that implementation includes `tracestate` support.

The existing Correlation-Context support implemented in this repository would be separately copied into the repository introduced above, to complete the set of W3C propagators in one location.   After #381, the existing Correlation-Context half becomes the [baggage propagator](https://github.com/open-telemetry/opentelemetry-go/pull/381/files#diff-30211045a0e22194418a069406247c64), so the two parts will have been pre-factored.

(cc @iredelmeier @sjkaris)

**Comment by jmacd** [comment]
Action item: reach out to W3C team and get a new repo? If not, we'll create a repo in the OTel org. @mwear who should we ask about getting a new repo created in the W3C github org?

**Comment by mwear** [comment]
@SergeyKanzhelev do you think this belongs in the W3C Github org, and if so, can you help create a repository for it?

**Comment by MrAlias** [comment]
One thing that was pointed out to me by a few members of the Distributed Tracing W3C working group is that the W3C is a standards body and they didn't know of any other instances (off-hand at least) where the W3C hosted implementations of the specifications. That being said, I'm still interested to hear what @SergeyKanzhelev has to say about the subject.

**Comment by jmacd** [comment]
Update: https://github.com/lightstep/tracecontext.go is being made private, the same code is available https://github.com/iredelmeier/tracecontext.go

**Comment by MrAlias** [comment]
This should probably act as a superset of #75

**Comment by MrAlias** [comment]
At this point, I propose to resolve this issue by consolidating a complete and documented W3C propagation library here in this repo that this project can use. The package should be kept internal and if there is demand we can then move it to an external repo or expose it here.

**Comment by SergeyKanzhelev** [comment]
> @SergeyKanzhelev do you think this belongs in the W3C Github org, and if so, can you help create a repository for it?

I doubt we will have any engineering force behind the W3C to support it.

**Comment by MadVikingGod** [comment]
It look like W3C doesn't have a library that we could rely on.  It also doesn't seem apparent that we could incorporate it now with 1.0 being out.

I think we should close this as won't do.

**Comment by MrAlias** [comment]
> It look like W3C doesn't have a library that we could rely on. It also doesn't seem apparent that we could incorporate it now with 1.0 being out.
> 
> I think we should close this as won't do.

Agreed.
